src/containerapp/azext_containerapp/java_component_decorator.py

```diff
@@ -214,3 +227,121 @@ def process_loaded_scg_route(self):
+            self.java_component_def["properties"]["configurations"] = configuration_list
+
+    def construct_payload(self):
+        self.set_up_compoment_type()
+        self.set_up_service_bindings()
+        self.set_up_unbind_service_bindings()
+        self.set_up_gateway_route()
+        self.set_up_replicas()
+        self.validate_configurations()
+        # --configuration
+        self.set_up_configurations()
+        # --set-configuration
+        self.set_up_set_configurations()
+
+
+class JavaComponentUpdateDecorator(BaseJavaComponentDecorator):
+    def __init__(self, cmd: AzCliCommand, client: Any, raw_parameters: Dict, models: str):
+        super().__init__(cmd, client, raw_parameters, models)
+        current_java_component_def = self.show()
+        if "configurations" not in current_java_component_def["properties"] or not current_java_component_def["properties"]["configurations"]:
+            self.java_component_def["properties"]["configurations"] = []
+        else:
+            self.java_component_def["properties"]["configurations"] = current_java_component_def["properties"]["configurations"]
+
+    def update(self):
+        try:
+            return self.client.update(
+                cmd=self.cmd, resource_group_name=self.get_argument_resource_group_name(),
+                environment_name=self.get_argument_environment_name(), name=self.get_argument_java_component_name(),
+                java_component_envelope=self.java_component_def,
+                no_wait=self.get_argument_no_wait())
+        except Exception as e:
+            handle_raw_exception(e)
+
+    def add_or_update_configurations(self, existing_configurations, new_configurations):
+        for new_configuration in new_configurations:
+            if "value" not in new_configuration:
+                raise ValidationError("Value must be provided for configuration {}.".format(new_configuration["propertyName"]))
+
+            # Check if updating existing configuration
+            is_existing = False
+            for existing_configuration in existing_configurations:
+                if existing_configuration["propertyName"].lower() == new_configuration["propertyName"].lower():
+                    is_existing = True
+                    existing_configuration["value"] = new_configuration["value"]
+
+            # If not updating existing configuration, add it as a new configuration
+            if not is_existing:
+                existing_configurations.append(new_configuration)
+
+    def remove_configurations(self, existing_configurations, remove_configurations):
+        remove_set = {config.lower() for config in remove_configurations}
+
+        new_configurations = [
+            config for config in existing_configurations
+            if config["propertyName"].lower() not in remove_set
+        ]
+
+        for config in remove_configurations:
+            if config.lower() not in {c["propertyName"].lower() for c in existing_configurations}:
```

Comment:
is propertyName in the configuration case sensitive?

Reply:
They are case-sensitive, and the backend enforces they only accept lowercase one

Reply:
They are case-sensitive, why you lower() them to do the comparison?

Reply:
you are right, we should not change the input, updated